daemons/fenced/fenced_commands.c

```diff
@@ -2385,32 +2385,31 @@ send_async_reply(async_command_t *cmd, const pcmk__action_result_t *result,
                  int pid, bool merged)
 {
     xmlNode *reply = NULL;
-    gboolean bcast = FALSE;
+    bool bcast = false;
 
     CRM_CHECK((cmd != NULL) && (result != NULL), return);
 
     reply = construct_async_reply(cmd, result);
 
-    // Only replies for certain actions are broadcast
-    if (pcmk__str_any_of(cmd->action, "metadata", "monitor", "list", "status",
-                         NULL)) {
-        crm_trace("Never broadcast '%s' replies", cmd->action);
+    // If target was also the originator, broadcast fencing results for it
+    if (!stand_alone && pcmk__is_fencing_action(cmd->action)
+        && pcmk__str_eq(cmd->origin, cmd->victim, pcmk__str_casei)) {
 
-    } else if (!stand_alone && pcmk__str_eq(cmd->origin, cmd->victim, pcmk__str_casei) && !pcmk__str_eq(cmd->action, "on", pcmk__str_casei)) {
-        crm_trace("Broadcast '%s' reply for %s", cmd->action, cmd->victim);
+        crm_trace("Broadcast '%s' result for %s (target was also originator)",
```

Comment:
for symmetry could consider moving that to where the other traces are.
would it be an issue having F_STONITH_MERGED pulled up front?
otherwise structure could be simplified without bcast doing everything together with sending. 
xml_trace moved to after sending (shouldn't alter message right) to see the final message of course.

Reply:
I don't follow. What other traces? I don't see how F_STONITH_MERGED relates to bcast; with this commit, bcast only controls the sending. What I was going for was create reply -> log reply -> send reply.

Reply:
Idea was to possibly go add merge dependent -> add bcast dependent & send -> log.
Might be easier to follow as it might not need bcast flag - but didn't consider additions that might break that benefit.

Reply:
Ah, I see -- that makes sense. There shouldn't be any problem with that. We could even log the result before constructing the reply.

Reply:
Added another commit on top